manifests/master/config.pp

```diff
@@ -29,6 +29,33 @@
   $restart_puppet              = true,
 ) inherits puppetdb::params {
 
+  # **WARNING**: Ugly hack to work around a yum bug with metadata parsing. This
+  # should not be copied, replicated or even looked at. In short, never rename
+  # your packages...
+  #
+  # With `yum` we can't have the termini package override the terminus package
+  # because that would prevent users from installing v2.3 of the terminus in
+  # PC1. We tried using a dummy terminus-3 metadata package which pulled in
+  # termini-3.latest as a dependency and put a requires terminus >= 3, <4. The
+  # idea was that doing `yum install puppetdb-termini-3.x.y-1.el7` would pull up
+  # the terminus package to the dummy 3 version, but `yum` has a bug which
+  # requires that both the dummy package and termini be installed in the same
+  # transaction, i.e. `yum install puppetdb-termini-3.x.y-1.el7
+  # puppetdb-terminus-3` which is impossible to do via Puppet.
+  #
+  # This will orphan some old terminus files (from pre-puppet-agent, i.e. puppet
+  # 3.x) that we're orphaned anyways and some of the new terminus files
```

Comment:
moreover, "this will orphan some terminus files we're orphaning" is a meaningless statement